NomaiGrandPrix/NomaiGrandPrix.cs

```diff
@@ -191,8 +208,13 @@ private void SpeedRunButton_OnClick()
 
         private void ResetRunButton_OnClick()
         {
-            SpeedrunState.SpawnPoint = GetRandomSpawnConfig(config => { return config.shouldSpawn; });
-            SpeedrunState.GoalPoint = GetRandomSpawnConfig(config => { return config.shouldGoal; });
+            //TODO: figure out a good place to consolidate this logic since it's duplicated 4 times.
+            SpeedrunState.SpawnPoint = GetRandomSpawnConfig(config => { 
+              return config.shouldSpawn && (config.area == Area.None || ModHelper.Config.GetSettingsValue<bool>($"Spawn{areaNameMap[config.area]}"));
```

Comment:
We could use [Enum.parse()](https://learn.microsoft.com/en-us/dotnet/api/system.enum.parse?view=net-7.0) here. Something like `$"Spawn{Enum.Parse(typeof(Area), config.area)}"`. That would let us get rid of the static map.

Reply:
good call. I'll add that and change the naming so that it works.

Reply:
fixed